### PR TITLE
Add special chars encoding tests for image and video tags

### PR DESCRIPTION
**Pull request by maoznir**

### CODE REVIEW

File: test/spec/tagspec.js [hunk 1]
```diff
@@ -190,6 +198,30 @@ describe("Cloudinary.VideoTag", function() {
     [videoTag, tag] = getTag(videoTag);
     return expect(tag).toBe(`<source src="${expected_url}.ogv" type="video/ogg">`);
   });
+  it("should generate video tag with encoded special chars", function() {
+    let movieId = `test's special < \"characters\" >`;
+    let encodedMovieId = encodeURIComponent(movieId);
```
Comment (asisayag2) [inline test/spec/tagspec.js]: @maoznir same here

File: test/spec/tagspec.js [hunk 2]
```diff
@@ -190,6 +198,30 @@ describe("Cloudinary.VideoTag", function() {
...
+    expect(tag).toBe(`<source src="${expected_url}.webm" type="video/webm">`);
+    [videoTag, tag] = getTag(videoTag);
+    expect(tag).toBe(`<source src="${expected_url}.mp4" type="video/mp4">`);
+    [videoTag, tag] = getTag(videoTag);
+    return expect(tag).toBe(`<source src="${expected_url}.ogv" type="video/ogg">`);
+  });
+
+  it("should create an image tag with encoded special chars", function() {
+    let imageId = `test's special < \"characters\" >`;
+    let encodedImageId = encodeURIComponent(imageId);
```
Comment (asisayag2) [inline test/spec/tagspec.js]: @maoznir and here

File: test/spec/tagspec.js [hunk 3]
```diff
@@ -86,6 +86,14 @@ describe("Cloudinary.ImageTag", function() {
     tag = new cloudinary.ImageTag('image_id', config).toHtml();
     return expect(tag).toBe(`<img src="${DEFAULT_UPLOAD_PATH}image_id">`);
   });
+  it("should create an image tag with encoded special chars", function() {
+    let imageId = `test's special < \"characters\" >`;
+    let encodedImageId = encodeURIComponent(imageId);
```
Comment (asisayag2) [inline test/spec/tagspec.js]: @maoznir, using the same function that the tag creation is using to create the expected value is problematic. 
If there's a bug in the function, this test will still pass.  
Please put the actual value you are expecting to get

File: test/spec/tagspec.js (hunk 1 above)
Comment (maoznir) [reply test/spec/tagspec.js]: done

File: test/spec/tagspec.js (hunk 2 above)
Comment (maoznir) [reply test/spec/tagspec.js]: done

File: test/spec/tagspec.js (hunk 3 above)
Comment (maoznir) [reply test/spec/tagspec.js]: done

File: test/spec/tagspec.js (hunk 2 above)
Comment (maoznir) [reply test/spec/tagspec.js]: this was duplicate

**Pull request merged**